lib/vintage_net_wizard/backend_server.ex

```diff
@@ -326,4 +336,16 @@ defmodule VintageNetWizard.BackendServer do
 
     priority_index + 1
   end
+
+  defp clean_config(%{key_mgmt: :wpa_psk} = config) do
+    Map.drop(config, [:psk])
+  end
+
+  defp clean_config(%{key_mgmt: :wpa_eap} = config) do
+    Map.drop(config, [:password])
+  end
+
+  defp clean_config(config) do
+    config
+  end
```

Comment:
I have a personal preference for an allow list approach. I.e., `key_mgmt: :wpa_psk` only allows `:ssid`, etc. That seems safer if the goal here is to avoid private info from leaking. The other route would be to erase all private information fields no matter what the key_mgmt setting is.

```suggestion
  defp clean_config(config) do
    Map.drop(config, [:psk, :password])
  end
```